docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+
+.Bean With Custom Qualifier Example
+[source,java]
+----
+@Superior <1>
+@ApplicationScoped
+public class SuperiorTranslator extends Translator {
+
+    String translate(String sentence) { 
+      // ...
+    }
+}
+----
+<1> `@Superior` is a https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#defining_qualifier_types[qualifier annotation, window="_blank"].
+
+This bean would be assignable to `@Inject @Superior Translator` and `@Inject @Superior SuperiorTranslator` but not in `@Inject Translator`.
+The reason is that `@Inject Translator` is automatically transformed to `@Inject @Default Translator` during typesafe resolution.
+And since our `SuperiorTranslator` does not declare `@Default` only the original `Translator` bean is assignable.
+
+== _Q: Looks good. What is the bean scope?_
+
+The scope of a bean determines the lifecycle of its instances, i.e. when and where an instance should be created and destroyed.
+
+NOTE: Every bean has exactly one scope.
+
+== _Q: What scopes can I actually use in my Quarkus application?_
+
+A: You can use all the built-in scopes mentioned by the specification except for `javax.enterprise.context.ConversationScoped`.
+
+[options="header",cols="1,1"]
+|===
+|Annotation   |Description   
+//----------------------
+|`@javax.enterprise.context.ApplicationScoped`   | A single bean instance is used for the application and shared among all injection points. The instance is created lazily, i.e. once a method is invoked upon the <<client_proxies, client proxy>>. 
+|`@javax.inject.Singleton`   | Just like `@ApplicationScoped` except that no client proxy is used. The instance is created eagerly.
+|`@javax.enterprise.context.RequestScoped`   | The bean instance is associated with the current _request_ (usually an HTTP request). 
+|`@javax.enterprise.context.Dependent`   | This is a pseudo-scope. There is no context object that holds all dependent instances. Instead, a dependent instace is a dependent object of the bean it is injected into.
+|`@javax.enterprise.context.SessionScoped`   | This scope is backed by an `javax.servlet.http.HttpSession` object. It's only available if `quarkus-undertow` extension is used.
+|===
+
+NOTE: There can be other custom scopes provided by Quarkus extension. For example, `qquarkus-narayana-jta` provides `javax.transaction.TransactionScoped`.
+
+[[client_proxies]]
+== _Q: I don't undestand the concept of client proxies._
+
+Indeed, the https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#client_proxies[client proxies, window="_blank"] could be hard to grasp but they provide some useful functionality.
+A client proxy is basically an object that delegates all method invocations to a target bean instance.
+It's a container construct that implements `io.quarkus.arc.ClientProxy` and extends the bean class.
+The client proxy instance is always injected instead of a contextual instance of a normal scoped bean (e.g. `@ApplicationScoped` and `@RequestScoped`).
```

Comment:
`contextual instance`: maybe find a more straightforward term: `instead of an object of the bean class`.

Reply:
`contextual instance` is a term used within CDI spec. I think it is good to introduce it so that the reader understands it if they decide to open the specification.

Reply:
Hm, "contextual instance" is actually a well-defined term from the spec. "object of the bean class" is not always true because for intercepted beans it's actually an instance of a container construct (bean class subclass). I cann add a link to the spec plus show a generated client proxy pseudocode. As I said it's hard to find the balance ;-)